### PR TITLE
refactor: Refactor main

### PR DESCRIPTION
**Pull request by kuznetsss**
For #442.

### CODE REVIEW

File: src/app/CliArgs.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    // clang-format on
+    po::positional_options_description positional;
+    positional.add("conf", 1);
+
+    po::variables_map parsed;
+    po::store(po::command_line_parser(argc, argv).options(description).positional(positional).run(), parsed);
+    po::notify(parsed);
+
+    if (parsed.count("help") != 0u) {
+        std::cout << "Clio server " << util::build::getClioFullVersionString() << "\n\n" << description;
```
Comment (cindyyan317) [inline src/app/CliArgs.cpp:57]: why two '\n' here?

Comment (kuznetsss) [reply src/app/CliArgs.cpp:57]: I just copied it from main.  I assume the empty line separates version from options:
```
  $ clio_server --help
  Clio server clio-20240724174205-442_Refactor_main-47b31ed5

  Options:
    -h [ --help ]                         print help message and exit
    -v [ --version ]                      print version and exit
    -c [ --conf ] arg (=/etc/opt/clio/config.json)
                                          configuration file
```
I don't mind removing it if you think it will be better without it.

File: src/web/Server.cpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2023, the clio developers.
```
Comment (cindyyan317) [inline src/web/Server.cpp]: 2024

File: src/web/Server.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+makeServerSslContext(util::Config const& config)
+{
+    bool const configHasCertFile = config.contains("ssl_cert_file");
+    bool const configHasKeyFile = config.contains("ssl_key_file");
+
+    if (configHasCertFile != configHasKeyFile) {
+        return std::unexpected{"Config entries 'ssl_cert_file' and 'ssl_key_file' must be set or unset together."};
+    }
+
+    if (not configHasCertFile) {
```
Comment (cindyyan317) [inline src/web/Server.cpp]: nit, should we remove {} here? I thought it is forced by clang-format

File: src/app/CliArgs.cpp (hunk 1 above)
Comment (godexsoft) [reply src/app/CliArgs.cpp:57]: Definitely not better without it 🔢

**Review by cindyyan317 — DISMISSED** [review]
Code looks good to me. But ut is failing.

File: src/web/Server.cpp (hunk 3 above)
Comment (kuznetsss) [reply src/web/Server.cpp]: Maybe clang-tidy would found it. Fixed.

**Review by cindyyan317 — DISMISSED** [review]

File: src/app/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+add_library(clio_app)
+target_sources(clio_app PRIVATE CliArgs.cpp ClioApplication.cpp)
+
+target_link_libraries(clio_app PUBLIC clio_etl clio_feed clio_web clio_rpc)
+
+# target_compile_features(clio_app PUBLIC cxx_std_23)
```
Comment (godexsoft) [inline src/app/CMakeLists.txt]: I assume we don't need this line?

File: src/app/ClioApplication.hpp [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    /**
+     * @brief Construct a new ClioApplication object
+     *
+     * @param config The configuration of the application
+     */
+    ClioApplication(util::Config const& config);
+
+    /**
+     * @brief Run the application
+-    *
```
Comment (godexsoft) [inline src/app/ClioApplication.hpp]: Looks like a stray `-`

File: src/util/requests/impl/SslContext.cpp [hunk 6]
```diff
@@ -82,9 +82,9 @@ getRootCertificate()
 }  // namespace
 
 std::expected<boost::asio::ssl::context, RequestError>
-makeSslContext()
+makeClientSslContext()
 {
-    ssl::context context{ssl::context::sslv23_client};
+    ssl::context context{ssl::context::tls_client};
```
Comment (godexsoft) [inline src/util/requests/impl/SslContext.cpp:87]: Seems unrelated to main refactoring.. just wanted to confirm that this is tested and works fine?

File: src/web/Server.hpp [hunk 7]
```diff
@@ -201,7 +211,7 @@ class Server : public std::enable_shared_from_this<Server<PlainSessionType, SslS
 
     util::Logger log_{"WebServer"};
     std::reference_wrapper<boost::asio::io_context> ioc_;
-    std::optional<std::reference_wrapper<boost::asio::ssl::context>> ctx_;
+    std::optional<boost::asio::ssl::context> ctx_;
```
Comment (godexsoft) [inline src/web/Server.hpp:214]: Are we sure this works as expected? i imagine it was previously a reference for a reason?

**Review by godexsoft — CHANGES_REQUESTED** [review]
Found some small things. Overall i like the changes 👍

File: src/app/CMakeLists.txt (hunk 4 above)
Comment (godexsoft) [reply src/app/CMakeLists.txt]: I don't see it added anywhere else. Please check we really don't need it

File: src/util/requests/impl/SslContext.cpp (hunk 6 above)
Comment (kuznetsss) [reply src/util/requests/impl/SslContext.cpp:87]: I checked, works fine.

File: src/web/Server.hpp (hunk 7 above)
Comment (kuznetsss) [reply src/web/Server.hpp:214]: It was a reference to be able to create a context outside of Server. It was created in main and passed by reference. Now Server owns the context.

File: src/app/CMakeLists.txt (hunk 4 above)
Comment (kuznetsss) [reply src/app/CMakeLists.txt]: It is defined for `clio_options` library, so I think here it was duplicated.

**Review by godexsoft — APPROVED** [review]
